### ISSUE TITLE
DHCP v6 Support

### ISSUE THREAD

**Issue by bmhughes** [sous-chefs/dhcp#118]
### :speaking_head: Foreword

This is more of a RFC rather than a feature request as such as I'd plan on implementing the functionality myself but wanted to get a consensus on how to proceed first before starting as it will be a fair amount of work.

## :person_frowning: Problem Statement

There is no support for managing an ISC DHCPv6 server.

## :grey_question: Possible Solution

**Add the functionality to this cookbook**

A lot of the current resources and recipes have v6 analogies and the server is ran by the same executable with the `-6` flag so it would make some sense in this manner to add the functionality to this cookbook.

## :arrow_heading_up: Describe alternatives you've considered

**Create a new cookbook**

If it is felt that it would be out-of-scope to add DHCPv6 support to this cookbook I'd look to create a new one instead.

## :heavy_plus_sign: Additional context

Add any other context about the problem here. e.g. related issues or existing pull requests.

**Comment by bmhughes** [comment]
I've been working through this and currently have basic functionality working but getting this far has raised several more points that would be good to clarify before I go further.

I feel it would be good to keep a similar functionality to the existing resources where there are v6 commons but as I dig through the cookbook I'm noticing a lot of things that are being done in a "weird" way to me only having started using chef around 2-3 years ago. I assume they are like this due to legacy limitations or perhaps some programming efficiency that passes me by (I am not a pro developer).

Case in point is the way that pools are presently passed and processed within the subnet resource. 

1. Why are they done with a block rather than a normal property which accepts a Hash/Array of Hash? 
2. Why is there then a shell "pool" resource that just validates the block input? Would this not be better served by a library function?
3. Why is there a library function use to generate the included list.conf file? Would an accumulator pattern template not be more chef-esk than looping the resource list in the `includes?` method?

All of these personally strike me a not very chefy ways of doing things but maybe it is just purely ignorance on my part hence looking to clarify the design patterns. If there is good reason to keep them then the v6 resources should probably follow the same pattern but if not then I don't see why they shouldn't be refactored to be more modern and less obtuse to chef/ruby newcomers.

**Comment by bmhughes** [comment]
Implemented via #133
